### ISSUE TITLE
x/exp/gorelease: add glossary-defined code next to each (in)compatible change

### ISSUE THREAD

**Issue by myitcv** [golang/go#37414]
### What version of Go are you using (`go version`)?

<pre>
$ go version
go version devel +151ccd4bdb Mon Feb 24 02:36:05 2020 +0000 linux/amd64
$ git rev-parse HEAD
7c80518d1cc79ffde9e571fe4fd281f321d36200
</pre>

### Does this issue reproduce with the latest release?

Yes

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env
GO111MODULE="on"
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/myitcv/.cache/go-build"
GOENV="/home/myitcv/.config/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/myitcv/gostuff"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/home/myitcv/gos"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/home/myitcv/gos/pkg/tool/linux_amd64"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/home/myitcv/gostuff/src/github.com/myitcv/govim/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build347856664=/tmp/go-build -gno-record-gcc-switches"
</pre></details>

### What did you do?

Ran `gorelease` against `govim`:

```
$ gorelease -base=v0.0.29
github.com/govim/govim
----------------------
Compatible changes:
- EventUser: added

github.com/govim/govim/cmd/govim/config
---------------------------------------
Incompatible changes:
- Config.ExperimentalTempModfile: removed
Compatible changes:
- CommandHighlightReferences: added
- Config.HighlightReferences: added
- Config.TempModfile: added
- EnvVarGoplsGOMAXPROCSMinusN: added
- FunctionWorkspaceSymbol: added
- HighlightReferences: added

github.com/govim/govim/testsetup
--------------------------------
Compatible changes:
- EnvStrictVimBufferLifecycle: added

github.com/govim/govim/testdriver
---------------------------------
Compatible changes:
- Config.Vim: added
- VimConfig: added

Suggested version: v0.1.0
```

### What did you expect to see?

Some sort of code/reference against each change that allows me to refer to a glossary for more detail. Very much in the same vein as [`staticcheck`'s codes for each analysis](https://staticcheck.io/docs/checks).

### What did you see instead?

A plain presentation of the changes that were compatible/incompatible. 

cc @jayconrod

**Comment by jayconrod** [comment]
Indeed, most if not all of the errors and diagnostics that gorelease reports should come with references to documentation.

Many of these errors are subtle and require some deeper explanation.
